src/utils.jl

```diff
@@ -25,3 +22,10 @@ function line_to_tree(line::Vector)
   end
   return [line_to_tree(line[1:(end - 1)]), line[end]]
 end
+
+function getindices_narrow_keytype(d::Dictionary, indices)
+  T = typeof(d)
+  d_at_indices = getindices(d, indices)
+  T != typeof(d_at_indices) && (d_at_indices = T(d_at_indices))
```

Comment:
I think you can just use `d_at_indices = convert(typeof(d), d_at_indices)`, which won't do anything if `typeof(d) == typeof(d_at_indices)`.

So we could just write this as:
```julia
function getindices_narrow_keytype(d::Dictionary, indices)
  return convert(typeof(d), getindices(d, indices))
end
```

Reply:
Thanks, that's good to know.